### DISCUSSION TITLE
Object poses in a Environment

### DISCUSSION THREAD

**Discussion by Moj-Dev** [microsoft/AirSim#3974] [Support Q&A]
I want to define some random position points in the Blocks environment and train a drone as an RL agent to reach to the goals. I should reject the goals which are in the object's mesh area. How can I check it and choose just the points that are not in the blocks?

**Comment by GimpelZhang** [comment]
Sizes and positions of all obstacles in the environment can be read through UE editor. So I think obtaining these information by hand can definitely solve this problem.

**Comment by Moj-Dev** [comment]
@GimpelZhang Thanks for your comment. 
For each episode (Run), I want to randomize the position of objects. So I need a command to get the position of object mesh in the environment.

**Comment by jonyMarino** [comment]
Hi! did you try with simSetObjectPose? (https://microsoft.github.io/AirSim/apis/#common-apis)
